split xwayland from xorg-server

paper42 opened the pull request:
<!-- Mark items with [x] where applicable -->

#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

<!--
If GitHub CI cannot be used to validate the build result (for example, if the
build is likely to take several hours), make sure to
[skip CI](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#continuous-integration).
When skipping CI, uncomment and fill out the following section.
Note: for builds that are likely to complete in less than 2 hours, it is not
acceptable to skip CI.
-->
<!-- 
#### Does it build and run successfully? 
(Please choose at least one native build and, if supported, at least one cross build. More are better.)
- [ ] I built this PR locally for my native architecture, (ARCH-LIBC)
- [ ] I built this PR locally for these architectures (if supported. mark crossbuilds):
  - [ ] aarch64-musl
  - [ ] armv7l
  - [ ] armv6l-musl
-->
xwayland releases were recently split from xorg releases. This PR works well on my system, but some parts should be improved.

@leahneukirchen you are the maintainer of xorg-server, do you want to maintain xwayland too? or @ericonr? (maintainer of wayland)

TODO:
- [x] fiigure out `depends=`
- [x] solve file conflict with xorg-server (`/usr/lib/xorg/protocol.txt`)

ericonr commented:
I would suggest naming the new package `xorg-server-xwayland` (so you just revbump xorg-server and remove the subpackage from there), it should make for the easiest transition. I'm not sure renaming to xwayland is necessary, but I can be convinced otherwise.

Is the new package picking up updates correctly? It might need an `update` file.

You can put me as maintainer, if you don't want to do it yourself.

Haven't looked at the contents of the conflicting file, but my gut reaction is to `rm` it from xwayland.

paper42 commented:
* renamed the package back to xorg-server-xwayland
* added a new subpackage xorg-server-common which provides the conflicting file (protocol.txt). This is neccessary because xorg-server-xwayland doesn't depend on xorg-server, but still needs the file.
* I will take the maintainership

ericonr commented on srcpkgs/xorg-server/template:
```diff
@@ -109,6 +102,13 @@ xorg-server-xvfb_package() {
 	}
 }
 
+xorg-server-common_package() {
```
If this is being split out, it should be in `depends` for `xorg-server`. But I'd like to confirm that this is necessary and it can't simply be removed from `Xwayland`, which would be much easier.

paper42 replied:
Other distributions (arch, alpine) split protocol.txt to a separate package, but I tested if xwayland works without this file and it seems to work fine. The source seems to suggest protocol.txt is only installed and never used. I removed the subpackage.

ericonr replied:
It seems to be used here: https://sources.debian.org/src/xorg-server/2:1.20.10-3/dix/registry.c/?hl=35#L35

paper42 replied:
Oh, thanks for catching this. This code path is hit sometimes, because we enable xcsecurity, I will add the -common subpackage back.
